src/inmanta/moduletool.py

```diff
@@ -251,17 +248,24 @@ def diff(cls, *, low: Version, high: Version) -> Optional["ChangeType"]:
             return cls.MINOR
         if high.micro > low.micro:
             return cls.PATCH
+        if len(high.base_version.split(".")) >= 4:
+            high_revision = high.base_version.split(".")[3]
+            # We are switching from 3 digits to 4
+            if len(low.base_version.split(".")) < 4 or high_revision > low.base_version.split(".")[3]:
```

Comment:
This doesn't look correct. You are doing a string comparison, but you actually want to do compare numbers.